src/ray/object_manager/object_buffer_pool.cc

```diff
@@ -327,8 +327,31 @@ ray::Status ObjectBufferPool::EnsureBufferExists(const ObjectID &object_id,
 }
 
 void ObjectBufferPool::FreeObjects(const std::vector<ObjectID> &object_ids) {
-  absl::MutexLock lock(&pool_mutex_);
-  RAY_CHECK_OK(store_client_->Delete(object_ids));
+  constexpr int kMaxAttempts = 3;
+  absl::Duration backoff = absl::Milliseconds(5);
+  int attempt = 1;
+  while (attempt <= kMaxAttempts) {
+    Status s;
+    {
+      absl::MutexLock lock(&pool_mutex_);
+      s = store_client_->Delete(object_ids);
```

Comment:
The lock was added in https://github.com/ray-project/ray/commit/3df1e1c471aef6a54d2c5011e0839f5fb9443922 but is it really required here? Per my understanding, `FreeObjects` doesn't read/modify `ObjectBufferPool` state (it only calls `store_client_->Delete`), and the plasma client is internally synchronized with its own mutex.

If we keep the lock, I would prefer to NOT lock across the entire retry loop. It'll serialize pool operations during the sleeps. Reacquiring `pool_mutex_` only around each `Delete` keeps other pool operations (create/abort) unblocked during backoff. Wdyt @edoakes @dayshah ?

Reply:
ok, i'll remove that in a separate PR

Reply:
https://github.com/ray-project/ray/pull/57833